src/ioctls/vcpu.rs

```diff
@@ -2375,6 +2527,146 @@ mod tests {
         }
     }
 
+    #[cfg(target_arch = "x86_64")]
+    #[test]
+    fn test_sync_regs() {
+        let kvm = Kvm::new().unwrap();
+        let vm = kvm.create_vm().unwrap();
+        let mut vcpu = vm.create_vcpu(0).unwrap();
+
+        vcpu.set_sync_valid_reg(SyncReg::Register);
+        assert_eq!(
+            vcpu.kvm_run_ptr.as_mut_ref().kvm_valid_regs,
+            SyncReg::Register as u64
+        );
+        vcpu.clear_sync_valid_reg(SyncReg::Register);
+        assert_eq!(vcpu.kvm_run_ptr.as_mut_ref().kvm_valid_regs, 0);
+
+        vcpu.set_sync_valid_reg(SyncReg::SystemRegister);
+        assert_eq!(
+            vcpu.kvm_run_ptr.as_mut_ref().kvm_valid_regs,
+            SyncReg::SystemRegister as u64
+        );
+        vcpu.clear_sync_valid_reg(SyncReg::SystemRegister);
+        assert_eq!(vcpu.kvm_run_ptr.as_mut_ref().kvm_valid_regs, 0);
+
+        vcpu.set_sync_valid_reg(SyncReg::VcpuEvents);
+        assert_eq!(
+            vcpu.kvm_run_ptr.as_mut_ref().kvm_valid_regs,
+            SyncReg::VcpuEvents as u64
+        );
+        vcpu.clear_sync_valid_reg(SyncReg::VcpuEvents);
+        assert_eq!(vcpu.kvm_run_ptr.as_mut_ref().kvm_valid_regs, 0);
```

Comment:
This code block and the ones below with `sync_dirty_reg` can each be reduced to the following (without loosing too much of the code readability):
```rust
        let sync_regs = [SyncReg::Register, SyncReg::SystemRegister, SyncReg::VcpuEvents];
        for reg in sync_regs {
            vcpu.set_sync_valid_reg(reg);
            assert_eq!(
                vcpu.kvm_run_ptr.as_mut_ref().kvm_valid_regs,
                SyncReg::Register as u64
            );
            vcpu.clear_sync_valid_reg(reg);
            assert_eq!(vcpu.kvm_run_ptr.as_mut_ref().kvm_valid_regs, 0);
        }
```

Would you be okay with this change to remove some of the code duplication?

Reply:
For sure. That makes total sense to me.